src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -1318,26 +1318,58 @@ public VirtualFile resolveOrLoadChildFromURL(URL url, boolean force, Predicate<S
             throws IOException {
         Watch watch = Watch.start();
         String path = parsePathFromUrl(url, fileExtensionVerifier);
-        boolean downloaded;
         Optional<LocalDateTime> lastModifiedHeader = Optional.empty();
 
         if (Strings.isEmpty(path)) {
-            Outcall outcall = new Outcall(url);
-            outcall.markAsHeadRequest();
-            Optional<String> contentDispositionPath = outcall.parseFileNameFromContentDisposition();
-            lastModifiedHeader = outcall.getHeaderFieldDate(HttpHeaderNames.LAST_MODIFIED.toString());
-
-            if (contentDispositionPath.isPresent() && fileExtensionVerifier.test(Files.getFileExtension(
-                    contentDispositionPath.get()))) {
-                path = contentDispositionPath.get();
-            } else {
-                throw Exceptions.createHandled()
-                                .withNLSKey("VirtualFile.loadFromUrl.noValidPath")
-                                .set("url", url)
-                                .handle();
+            Tuple<Optional<String>, Optional<LocalDateTime>> pathAndModifiedHeaders = requestHeaders(url);
```

Comment:
where is the 2nd part of the tuple used?

Reply:
In the `doDownload` method below